ls() crashes once directory listing has expired

When `listings_expiry_time` is set and the listing expires, `ls` crashes with 

```
[07/Sep/2022 13:28:51] [ERROR] [task]    File "/usr/local/lib/python3.7/site-packages/s3fs/core.py", line 1010, in _info
[07/Sep/2022 13:28:51] [ERROR] [task]      out = [o for o in out if o["name"] == path]
[07/Sep/2022 13:28:51] [ERROR] [task]  TypeError: 'NoneType' object is not iterable
```

because `_ls_from_cache` returns None

--- a/s3fs/core.py
+++ b/s3fs/core.py
@@ -1158,12 +1158,13 @@ class S3FileSystem(AsyncFileSystem):
         if path in ["/", ""]:
             return {"name": path, "size": 0, "type": "directory"}
         version_id = _coalesce_version_id(path_version_id, version_id)
-        if not refresh and self._ls_from_cache(path) is not None:
+        if not refresh:
             out = self._ls_from_cache(path)
-            out = [o for o in out if o["name"] == path]
-            if out:
-                return out[0]
-            return {"name": path, "size": 0, "type": "directory"}
+            if out is not None:
+                out = [o for o in out if o["name"] == path]
+                if out:
+                    return out[0]
+                return {"name": path, "size": 0, "type": "directory"}
         if key:
             try:
                 out = await self._call_s3(

